src/setters/depowithdraw.ts

```diff
@@ -44,11 +44,11 @@ export const setDepoWithdrawTx = (
     tx.userAddress = ev.userAddress;
     tx.fromAddress = ev.fromAddress;
     tx.toAddress = ev.toAddress;
-    tx.coinAmount = getCoinAmount(logs, tx, false);
+    tx.coinAmount = ev.coinAmount.isZero() 
+       ? getCoinAmount(logs, tx, false) 
+       : tokenToDecimal(ev.coinAmount, 18, DECIMALS);
     tx.usdAmount = tokenToDecimal(ev.usdAmount, 18, DECIMALS);
-    tx.factor = (token === 'pwrd')
-        ? getStoredFactor(token)
-        : getFactor(token);
+    tx.factor = getFactor(token);
```

Comment:
This is a **breaking change**. With the current protocol, the way to get the correct pwrd based amount is by using the factor just **before** the withdrawal, and not the one in the current block (after the withdrawal).
If this is different for G2, then we should add and `if` so that we call `getStoredFactor()` for pwrd if the block number is before the G2 deployment.
For further info, see this discord msg and the below answers: https://discord.com/channels/748111918698463312/943797208329883689/1034460634001186966